xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1659,26 +1659,27 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
         int size = 0;
         uint8_t* side_data = nullptr;
 
-        side_data = av_stream_get_side_data(pStream, AV_PKT_DATA_DOVI_CONF, &size);
-        if (side_data && size)
+        if (av_stream_get_side_data(pStream, AV_PKT_DATA_DOVI_CONF, &size) && size)
           st->hdr_type = StreamHdrType::HDR_TYPE_DOLBYVISION;
+        else if (av_stream_get_side_data(pStream, AV_PKT_DATA_DYNAMIC_HDR10_PLUS, &size) && size)
+          st->hdr_type = StreamHdrType::HDR_TYPE_HDR10PLUS;
         else if (st->colorPrimaries == AVCOL_PRI_BT2020)
         {
+          side_data =
```

Comment:
This is wrong. You are changing current behavior: `side_data` should be independent of if primaries is `AVCOL_PRI_BT2020` or not.

Reply:
You're correct will fix.